### ISSUE TITLE
aarch64: syscall fstatat gives wrong file path

### ISSUE THREAD

**Issue by shrik3** [QuarkContainer/Quark#1195]
reproduce with `busybox` container, `sh` then `ls`.

The fstatat returned path is cutoff by one leading char, e.g. `/etc` becomes `/tc` and `/home` becomes `/ome`

```
[INFO] [2/43c0480000|0] (2/2)------get call id sys_write arg0:2
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is 2b: call id sys_write 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_newfstatat arg0:ffffffffffffff9c
[INFO] [2/43c0480000|0] Fstatat path is ./8dd5bb6-9da9-4f5f-bcb0-316829ff718b dirPath false, flags & ATType::AT_SYMLINK_NOFOLLOW 100
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is fffffffffffffffe: call id sys_newfstatat 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_write arg0:2
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is 45: call id sys_write 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_newfstatat arg0:ffffffffffffff9c
[INFO] [2/43c0480000|0] Fstatat path is ./in dirPath false, flags & ATType::AT_SYMLINK_NOFOLLOW 100
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is fffffffffffffffe: call id sys_newfstatat 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_write arg0:2
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is 24: call id sys_write 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_newfstatat arg0:ffffffffffffff9c
[INFO] [2/43c0480000|0] Fstatat path is ./ev dirPath false, flags & ATType::AT_SYMLINK_NOFOLLOW 100
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is fffffffffffffffe: call id sys_newfstatat 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_write arg0:2
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is 24: call id sys_write 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_newfstatat arg0:ffffffffffffff9c
[INFO] [2/43c0480000|0] Fstatat path is ./tc dirPath false, flags & ATType::AT_SYMLINK_NOFOLLOW 100
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is fffffffffffffffe: call id sys_newfstatat 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_write arg0:2
[INFO] [2/43c0480000|0] (2/2)------Return[0] res is 24: call id sys_write 
[INFO] [2/43c0480000|0] (2/2)------get call id sys_newfstatat arg0:ffffffffffffff9c
[INFO] [2/43c0480000|0] Fstatat path is ./ome dirPath false, flags & ATType::AT_SYMLINK_NOFOLLOW 100
```

**Comment by shrik3** [comment]
![image](https://github.com/QuarkContainer/Quark/assets/16338681/21e06a43-c644-4de5-9856-be800f29d25d)

sys_gentdents64 is fine. I guess the problem is with fstat

**Comment by shrik3** [comment]
Okay, I see the problem.

The `struct dirent` returned by the host syscall could contain either or not the `char d_type` field. If you have a wrong assumption about the struct, reading the `d_name` field will have an off-by-one.

For linux kernel and glibc, this is checked with `_DIRENT_HAVE_D_TYPE`. But  I don't know yet how to set this up in our case. 

Not so obviously, our arm server with 5.15 kernel spits `dirent` WITHOUT the `d_type` field. Hence the problem in this issue.

**Comment by shrik3** [comment]
![image](https://github.com/QuarkContainer/Quark/assets/16338681/84f2bed4-85b2-4389-a5e6-7ab4c3200621)


So, the qkernel delegated syscall somhow invoked the host sys_getdents64 to use the old format (i.e. for x86 sys_gentdents).

**Comment by shrik3** [comment]
fixed #1201
